system/Commands/Generators/MigrationGenerator.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/**
+ * This file is part of the CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Commands\Generators;
+
+use CodeIgniter\CLI\BaseCommand;
+use CodeIgniter\CLI\CLI;
+use CodeIgniter\CLI\GeneratorTrait;
+
+/**
+ * Generates a skeleton migration file.
+ */
+class MigrationGenerator extends BaseCommand
+{
+	use GeneratorTrait;
+
+	/**
+	 * The Command's Group
+	 *
+	 * @var string
+	 */
+	protected $group = 'Generators';
+
+	/**
+	 * The Command's Name
+	 *
+	 * @var string
+	 */
+	protected $name = 'make:migration';
+
+	/**
+	 * The Command's Description
+	 *
+	 * @var string
+	 */
+	protected $description = 'Generates a new migration file.';
+
+	/**
+	 * The Command's Usage
+	 *
+	 * @var string
+	 */
+	protected $usage = 'make:migration <name> [options]';
+
+	/**
+	 * The Command's Arguments
+	 *
+	 * @var array
+	 */
+	protected $arguments = [
+		'name' => 'The migration class name.',
+	];
+
+	/**
+	 * The Command's Options
+	 *
+	 * @var array
+	 */
+	protected $options = [
+		'--session'   => 'Generates the migration file for database sessions.',
+		'--namespace' => 'Set root namespace. Defaults: "APP_NAMESPACE".',
+		'--suffix'    => 'Append the component title to the class name (e.g. UserMigration => User).',
+	];
+
+	/**
+	 * Actually execute a command.
+	 *
+	 * @param array $params
+	 */
+	public function run(array $params)
+	{
+		$this->component = 'Migration';
+		$this->directory = 'Database\Migrations';
+		$this->template  = 'migration.tpl.php';
+
+		if ($this->getOption('session'))
+		{
+			$params[0] = '_create_ci_session_table';
+		}
```

Comment:
Not sure what the exact problem is, but in `testGenerateMigrationWithOptionSession` this condition is not fulfilled. 
If you do `make:migration test -session` it works (but the test fails of course).
So there might be a proplem with recognizing the `--session` option with `getOption`.

For some reason `$this->getOption('session')` is `bool(true)` in the `prepare` method but it is `null` in the `run` method.

Reply:
The issue seems to be that when calling `$this->getOptions`, the `$params` array is not yet copied to `$this->params` as that only happens in the `execute` function. I feel like that call should thus move to `prepare` or you just directly check the `$params` array in the run function for the `session` index.

Reply:
Oh what a suffering from that problem, looks like everything is working fine now, I appreciate your review and help Thank you @sfadschm